minwoo.seo/src/BJ2564.java

```diff
@@ -0,0 +1,71 @@
+import java.util.Scanner;
+
+public class BJ2564 {
+  public static void main(String[] args) {
+    Scanner sc = new Scanner(System.in);
+    int m = sc.nextInt(); // 가로
+    int n = sc.nextInt(); // 세로
+
+    int s = sc.nextInt();
+    Point[] stores = new Point[s];
+
+    for (int i = 0; i < s; i++) {
+      stores[i] = new Point(sc.nextInt(), sc.nextInt(), n, m);
+    }
+    Point dongeun = new Point(sc.nextInt(), sc.nextInt(), n, m);
+
+    int side = checkSide(dongeun.direction);
+    int result = 0;
+    for (Point store : stores) {
+      if (store.direction == side) {
+        if (store.direction == 1 || store.direction == 2) {
+          result += Math.min(dongeun.y + n + store.y, (m - dongeun.y) + n + (m - store.y));
+        } else {
+          result += Math.min(dongeun.x + m + store.x, (n - dongeun.x) + m + (n - store.x));
+        }
+      } else {
+        result += Math.abs(dongeun.x - store.x) + Math.abs(dongeun.y - store.y);
+      }
+    }
+    System.out.println(result);
+  }
+
+  private static int checkSide(int me) {
+    if (me == 1) {
+      return 2;
+    } else if (me == 2) {
+      return 1;
+    } else if (me == 3) {
+      return 4;
+    } else {
+      return 3;
+    }
```

Comment:
수식을 이용하면 이렇게 바꿀수도 있습니당
```suggestion
    return (me + 1) / 2 + (me - 1) / 2 + me % 2;
```

Reply:
이 수식을 떠올리는게 더 어려워보이는데요... 이건 어떻게 떠올리는거죠 ㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋ